com.unity.netcode.gameobjects/Samples/ClientNetworkAnimator/Scripts/ClientNetworkAnimator.cs

```diff
@@ -0,0 +1,13 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// A prototype component for syncing Mecanim Animator state in a client-driven manner
+    /// </summary>
+    [AddComponentMenu("Netcode/" + nameof(ClientNetworkAnimator))]
+    public class ClientNetworkAnimator : NetworkAnimator
+    {
+        public override bool IsAuthorityOverAnimator => IsClient && IsOwner;
```

Comment:
should be protected not public.
also from internal discussions, this should have a big comment saying you can't update server network animator by default and to use an RPC to update it (On NetworkAnimator itself)
in here, should have same comment, that this works because we're using RPCs

Reply:
You can't update NetworkAnimator (Server or Client authoritative) directly period, it has no interface for that.

 It will poll the the animator (if you're the authority over it, depending on the implementation) and sync out it's values - if another client decides to just do random stuff to the Mecanim animator - he can do that, but the change would be overriden as soon as the data gets delivered from the server. 

The usage example of this code can be found in the manual test called NetworkAnimatorTest, the class in question is `AnimatedCubeController` - there it uses the value of the `IsAuthorityOverAnimator` to decide to set the triggers directly or to invoke an RPC that would make the server do that.

Reply:
What is the value of making it protected instead of public? There's a practical reason for keeping it public - I want the user to be able to check if this particular NetworkAnimator has authority in order to determine how to do input. If it's protected - our users will have to reimplement the same checks if they need to choose between changing the values directly or an RPC.

Reply:
resolving after mine and Sam's discussion offline - renamed this "authority" bit to `WillCommitChanges` - seems like a better fit for what it does.